src/EnergyPlus/ScheduleManager.cc

```diff
@@ -4491,6 +4491,55 @@ namespace ScheduleManager {
         return MaximumValue;
     }
 
+    std::pair<Real64, Real64> getScheduleMinMaxByDayType(EnergyPlusData &state, int const ScheduleIndex, DayTypeGroup const days)
+    {
+        // J. Glazer - March 2024
+        // finds the minimum and maximum for a specific set of day types for a given schedule
+        Real64 MinValue = Constant::BigNumber;
+        Real64 MaxValue = -Constant::BigNumber;
+        if (ScheduleIndex == -1) {
+            MinValue = 1.0;
+            MaxValue = 1.0;
+        } else if (ScheduleIndex == 0) {
+            MinValue = 0.0;
+            MaxValue = 0.0;
+        } else if (ScheduleIndex < 1 || ScheduleIndex > state.dataScheduleMgr->NumSchedules) {
+            ShowFatalError(state, "getScheduleMinMaxByDayType called with ScheduleIndex out of range");
+        } else if (ScheduleIndex > 0) {
+            std::vector<bool> dayTypeFilter(maxDayTypes);
+            std::fill(dayTypeFilter.begin(), dayTypeFilter.end(), false);
+            switch (days) {
+            case DayTypeGroup::Weekday:
+                //               Sun    Mon   Tues  Wed   Thur  Fri   Sat    Sun    Summer Winter Cust1  Cust2
+                dayTypeFilter = {false, true, true, true, true, true, false, false, false, false, false, false};
+                break;
+            case DayTypeGroup::WeekEndHoliday:
+                dayTypeFilter = {true, false, false, false, false, false, true, true, false, false, false, false};
+                break;
```

Comment:
The first true is Sunday, then five falses are Monday to Friday, then true for Saturday and then true for Holiday.

Reply:
I see, my comment shows Sunday twice. I should probably verify which one is really Sunday :-)

Reply:
The other thing I noticed is that this dayTypeFilter array is a constexpr for each day type. It would be nice to set these up at compile time and somehow use the correct array in the loop below.

Reply:
I fixed the comment based on the enumeration:

https://github.com/NREL/EnergyPlus/blob/895537498b0faa14e8574a5444cc57b3e89f87c0/src/EnergyPlus/ScheduleManager.hh#L73-L90